twoliter/src/tools.rs

```diff
@@ -0,0 +1,78 @@
+use crate::common::fs;
+use crate::tools_hash::TOOLS_HASH;
+use anyhow::{Context, Result};
+use flate2::read::GzDecoder;
+use log::{debug, trace, warn};
+use std::path::Path;
+use tar::Archive;
+
+const TARBALL_DATA: &[u8] = include_bytes!(concat!(env!("OUT_DIR"), "/tools.tar.gz"));
+
+pub(crate) async fn install_tools(tools_dir: impl AsRef<Path>, force: bool) -> Result<()> {
+    let tools_dir = tools_dir.as_ref();
+    let sentinel_filepath = tools_dir.join("installed");
+    if !force && !should_install(&sentinel_filepath).await {
+        debug!("Not installing tools because hashes matched");
+        return Ok(());
+    }
+
+    trace!("Installing tools to '{}'", tools_dir.display());
+
+    fs::create_dir_all(&tools_dir)
+        .await
+        .context("Unable to create a directory for Twoliter's tools")?;
+
+    // Write out the embedded tools and scripts.
+    unpack_tarball(&tools_dir)
+        .await
+        .context("Unable to install tools")?;
+
+    // Write out a file that can be used to detect what version of the tools has been installed.
+    fs::write(&sentinel_filepath, &TOOLS_HASH)
+        .await
+        .context(format!(
+            "Unable to write the tools hash to '{}'",
+            sentinel_filepath.display(),
+        ))?;
+
+    Ok(())
+}
+
+async fn should_install(sentinel_filepath: &Path) -> bool {
```

Comment:
```suggestion

/// Returns `false` if the tools hash file is found and contains the same hash as `TOOLS_HASH`.
/// Otherwise returns `true` because the tools have not been installed or may not be the correct
/// version. 
async fn should_install(sentinel_filepath: &Path) -> bool {
```